### PR TITLE
Fix AI steering spamming console

### LINKED ISSUE #1251: JobQueue exceptions
My brain too smol to fix.

```
[ERRO] job: Job failed on exception:
System.OperationCanceledException: The operation was canceled.
   at System.Threading.CancellationToken.ThrowOperationCanceledException()
   at System.Threading.CancellationToken.ThrowIfCancellationRequested()
   at Content.Server.GameObjects.EntitySystems.JobQueues.Job`1.ProcessWrap() in D:\ProgrammingProjects\space-station-14\Content.Server\GameObjects\EntitySystems\JobQ
ueues\Job.cs:line 167
[ERRO] unobserved: System.AggregateException: A Task's exception(s) were not observed either by Waiting on the Task or accessing its Exception property. As a result,
 the unobserved exception was rethrown by the finalizer thread. (The operation was canceled.)
 ---> System.OperationCanceledException: The operation was canceled.
   at System.Threading.CancellationToken.ThrowOperationCanceledException()
   at System.Threading.CancellationToken.ThrowIfCancellationRequested()
   at Content.Server.GameObjects.EntitySystems.JobQueues.Job`1.ProcessWrap() in D:\ProgrammingProjects\space-station-14\Content.Server\GameObjects\EntitySystems\JobQ
ueues\Job.cs:line 167
   --- End of inner exception stack trace ---
```

Steps to reproduce (at least for me).
1. Spawn a lot of AI

It's coming from the pathfinding job as they're the ones that are cancelled the most.

### PR DESCRIPTION
**Pull request by metalgearsloth**
Turns out you shouldn't cancel pending jobs.

Should fix #1251 and maybe #1252

### CODE REVIEW

File: Content.Server/GameObjects/EntitySystems/AI/Steering/AiSteeringSystem.cs [hunk 1]
```diff
@@ -135,7 +135,32 @@ public void Unregister(IEntity entity)
...
+                        request.Item1.Cancel();
+                        break;
+                    case JobStatus.Waiting:
+                        request.Item1.Cancel();
+                        break;
+                    case JobStatus.Finished:
+                        break;
+                    default:
+                        throw new ArgumentOutOfRangeException();
+                }
```
Comment (RemieRichards) [inline Content.Server/GameObjects/EntitySystems/AI/Steering/AiSteeringSystem.cs:155]:
```suggestion
                switch (request.Item2.Status)
                {
                    case JobStatus.Pending:
                    case JobStatus.Finished:
                        break;
                    case JobStatus.Running:
                    case JobStatus.Paused:
                    case JobStatus.Waiting:
                        request.Item1.Cancel();
                        break;
                }
```

Comment (RemieRichards) [reply Content.Server/GameObjects/EntitySystems/AI/Steering/AiSteeringSystem.cs]: Also, we have C# 7.0, so these request tuples should really be named.

```cs
private readonly Dictionary<IEntity, (CancellationTokenSource, Job<Queue<TileRef>>)> _pathfindingRequests = 
    new Dictionary<IEntity, (CancellationTokenSource, Job<Queue<TileRef>>)>();
```
Should become
```cs
private readonly Dictionary<IEntity, (CancellationTokenSource CancelToken, Job<Queue<TileRef>> job)> _pathfindingRequests = 
    new Dictionary<IEntity, (CancellationTokenSource CancelToken, Job<Queue<TileRef>> job)>();
```

Which would let you do `request.CancelToken.cancel()` and `request.Job.Status`, I'm sure you can agree that's much more readable!

**Comment by RemieRichards** [comment]
(Don't forget to check out my other comment, it got folded with the previous comment)

File: Content.Server/GameObjects/EntitySystems/AI/Steering/AiSteeringSystem.cs (hunk 1 above)
Comment (metalgearsloth) [reply Content.Server/GameObjects/EntitySystems/AI/Steering/AiSteeringSystem.cs]: TIL, thanks remie. I definitely needed that for my next AI PR...

**Pull request merged**